src/App.tsx

```diff
@@ -63,6 +67,25 @@ function App() {
 
   const [state, dispatch] = useReducer(reducer, initialState);
 
+  const {cameraList, micList, selectedCamera, selectedMic, setCamera, setMic, mediaStream} = useMedia();
+
+  const video = useRef<HTMLVideoElement>(null)
+
+  useEffect(() => {
+    if (video?.current && (mediaStream instanceof MediaStream)) {
+      video.current.srcObject = mediaStream;
+    }
+  }, [mediaStream])
+
+  const setCamHandler = (groupId: string) => {
+    const device = cameraList.filter(cam => cam.groupId === groupId)
+    setCamera(device[0]);
+  }
+
+  const setMicHandler = (groupId: string) => {
```

Comment:
same as setCameraHandler above

Reply:
also `setMicrophoneHandler` as we plan on not using `mic` anywhere.